Description: Move general info man pages to the miscellanea section (7)
 This patch moves the general information man pages slurm.1 and spank.8
 to the miscellanea section since they are not related to commands
Author: Gennaro Oliva <oliva.g@na.icar.cnr.it>
Forwarded: https://bugs.schedmd.com/show_bug.cgi?id=8719
Last-Update: 2020-11-05

diff --git a/configure.ac b/configure.ac
index e1aff79c3d..c8512a2c2c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -405,6 +405,7 @@ AC_CONFIG_FILES([Makefile
 		 doc/man/man1/Makefile
 		 doc/man/man3/Makefile
 		 doc/man/man5/Makefile
+		 doc/man/man7/Makefile
 		 doc/man/man8/Makefile
 		 doc/html/Makefile
 		 doc/html/configurator.html
diff --git a/doc/man/Makefile.am b/doc/man/Makefile.am
index 5a3ddf7f1f..e3afc1da90 100644
--- a/doc/man/Makefile.am
+++ b/doc/man/Makefile.am
@@ -1,3 +1,3 @@
 
-SUBDIRS = man1 man3 man5 man8
+SUBDIRS = man1 man3 man5 man7 man8
 
diff --git a/doc/man/man1/Makefile.am b/doc/man/man1/Makefile.am
index 710d12f6c8..c285a4e4ef 100644
--- a/doc/man/man1/Makefile.am
+++ b/doc/man/man1/Makefile.am
@@ -11,7 +11,6 @@ man1_MANS = \
 	scontrol.1 \
 	sdiag.1	\
 	sinfo.1   \
-	slurm.1 \
 	sprio.1 \
 	squeue.1 \
 	sreport.1 \
diff --git a/doc/man/man5/slurm.conf.5 b/doc/man/man5/slurm.conf.5
index 6958af8d76..dca152a3f4 100644
--- a/doc/man/man5/slurm.conf.5
+++ b/doc/man/man5/slurm.conf.5
@@ -1604,7 +1604,7 @@ This provides support for a highly configurable set of plugins to
 be called before and/or after execution of each task spawned as
 part of a user's job step.  Default location is "plugstack.conf"
 in the same directory as the system slurm.conf. For more information
-on SPANK plugins, see the \fBspank\fR(8) manual.
+on SPANK plugins, see the \fBspank\fR(7) manual.
 
 .TP
 \fBPowerParameters\fR
@@ -5873,4 +5873,4 @@ details.
 \fBgetrlimit\fR (2), \fBgres.conf\fR(5), \fBgroup\fR (5), \fBhostname\fR (1),
 \fBscontrol\fR(1), \fBslurmctld\fR(8), \fBslurmd\fR(8),
 \fBslurmdbd\fR(8), \fBslurmdbd.conf\fR(5), \fBsrun(1)\fR,
-\fBspank(8)\fR, \fBsyslog\fR (2), \fBtopology.conf\fR(5)
+\fBspank(7)\fR, \fBsyslog\fR (2), \fBtopology.conf\fR(5)
diff --git a/doc/man/man7/Makefile.am b/doc/man/man7/Makefile.am
new file mode 100644
index 0000000000..ba18a765f1
--- /dev/null
+++ b/doc/man/man7/Makefile.am
@@ -0,0 +1,22 @@
+htmldir = ${datadir}/doc/${PACKAGE}-${SLURM_VERSION_STRING}/html
+
+man7_MANS = slurm.7 \
+	spank.7
+
+EXTRA_DIST = $(man7_MANS)
+
+if HAVE_MAN2HTML
+
+html_DATA = \
+	spank.html
+
+MOSTLYCLEANFILES = ${html_DATA}
+
+EXTRA_DIST += $(html_DATA)
+
+SUFFIXES = .html
+
+.7.html:
+	`dirname $<`/../man2html.py @SLURM_MAJOR@.@SLURM_MINOR@ $(srcdir)/../../html/header.txt $(srcdir)/../../html/footer.txt $<
+
+endif
diff --git a/doc/man/man1/slurm.1 b/doc/man/man7/slurm.7
similarity index 97%
rename from doc/man/man1/slurm.1
rename to doc/man/man7/slurm.7
index ecb5617f9e..a043fb737b 100644
--- a/doc/man/man1/slurm.1
+++ b/doc/man/man7/slurm.7
@@ -1,4 +1,4 @@
-.TH Slurm "1" "Slurm System" "June 2018" "Slurm System"
+.TH Slurm "7" "Slurm System" "June 2018" "Slurm System"
 
 .SH "NAME"
 Slurm \- Slurm Workload Manager overview.
@@ -68,5 +68,5 @@ details.
 \fBsview\fR(1),
 \fBslurm.conf\fR(5), \fBslurmdbd.conf\fR(5),
 \fBslurmctld\fR(8), \fBslurmd\fR(8), \fBslurmdbd\fR(8), \fBslurmstepd\fR(8),
-\fBspank\fR(8)
+\fBspank\fR(7)
 
diff --git a/doc/man/man8/spank.8 b/doc/man/man7/spank.7
similarity index 99%
rename from doc/man/man8/spank.8
rename to doc/man/man7/spank.7
index 6141f49e6f..9be38511ba 100644
--- a/doc/man/man8/spank.8
+++ b/doc/man/man7/spank.7
@@ -1,4 +1,4 @@
-.TH SPANK "8" "Slurm Component" "April 2020" "Slurm Component"
+.TH SPANK "7" "Slurm Component" "April 2020" "Slurm Component"
 
 .SH "NAME"
 \fBSPANK\fR \- Slurm Plug\-in Architecture for Node and job (K)control
diff --git a/doc/man/man8/Makefile.am b/doc/man/man8/Makefile.am
index ef280ae95f..9aebfb399c 100644
--- a/doc/man/man8/Makefile.am
+++ b/doc/man/man8/Makefile.am
@@ -4,8 +4,7 @@ man8_MANS = slurmctld.8 \
 	slurmd.8 \
 	slurmdbd.8 \
 	slurmrestd.8 \
-	slurmstepd.8 \
-	spank.8
+	slurmstepd.8
 
 EXTRA_DIST = $(man8_MANS)
 
@@ -16,8 +15,7 @@ html_DATA = \
 	slurmd.html \
 	slurmdbd.html \
 	slurmrestd.html \
-	slurmstepd.html \
-	spank.html
+	slurmstepd.html
 
 MOSTLYCLEANFILES = ${html_DATA}
 
